stdlib/public/core/MutableCollection.swift

```diff
@@ -372,6 +372,30 @@ extension MutableCollection {
   }
 }
 
+extension MutableCollection where Self: BidirectionalCollection {
+  public subscript(bounds: Range<Index>) -> MutableBidirectionalSlice<Self> {
+    get {
+      _failEarlyRangeCheck(bounds, bounds: startIndex..<endIndex)
+      return .init(base: self, bounds: bounds)
+    }
+    set {
+      _writeBackMutableSlice(&self, bounds: bounds, slice: newValue)
+    }
+  }
+}
+
+extension MutableCollection where Self: RandomAccessCollection {
+  public subscript(bounds: Range<Index>) -> MutableRandomAccessSlice<Self> {
+    get {
+      _failEarlyRangeCheck(bounds, bounds: startIndex..<endIndex)
+      return .init(base: self, bounds: bounds)
```

Comment:
@airspeedswift The reason why repeat it is that despite years of Swift I *still* think of `.xxx` as looking up `xxx` in the current scope instead of the type context.  It feels as though it's chaining to `Self.init`.  Of course I'm projecting, but I have a hard time seeing how other people don't make the same mistake.  Finally, I don't see how it clarifies anything useful: `CapitalizedName(...)` is always creating something.  

@jckarter how does "gyb-ability" come into play?